src/Service/Services/GraphQLSchemaCreator.cs

```diff
@@ -211,18 +211,8 @@ DatabaseType.postgresql or
 
         private (DocumentNode, Dictionary<string, InputObjectTypeDefinitionNode>) GenerateCosmosGraphQLObjects()
         {
-            string graphqlSchema = ((CosmosSqlMetadataProvider)_sqlMetadataProvider).GraphQLSchema();
-
-            if (string.IsNullOrEmpty(graphqlSchema))
-            {
-                throw new DataApiBuilderException(
-                    message: "No GraphQL object model was provided for CosmosDB. Please define a GraphQL object model and link it in the runtime config.",
-                    statusCode: System.Net.HttpStatusCode.InternalServerError,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
-            }
-
             Dictionary<string, InputObjectTypeDefinitionNode> inputObjects = new();
-            DocumentNode root = Utf8GraphQLParser.Parse(graphqlSchema);
+            DocumentNode root = ((CosmosSqlMetadataProvider)_sqlMetadataProvider).ParseSchemaGraphQLDocument();
```

Comment:
We have the opportunity to optimize even further. Since GraphQLSchemaCreator is dependent on CosmosSqlMetadataProvider, whose constructor already calls `ParseSchemaGraphQLDocument()`. After parsing we could store the parsed DocumentNode _root as a member of the _sqlMetadataProvider and reuse that stored _root here so we dont need to reparse and call this same function again

Reply:
I refactored this and checking the root is null. 

If the root is null, then it throws a exception for being invalid schema. I wonder if I should return ```new()``` instead and just have no schema generated. Thoughts? @Aniruddh25

Reply:
Exception sounds good to me, so it doesnt silently go through and barf somewhere else. Wonder what was happening previously..